tempodb/encoding/vparquet3/block_traceql.go

```diff
@@ -1209,6 +1224,35 @@ func createSpanIterator(makeIter makeIterFn, primaryIter parquetquery.Iterator,
 		case traceql.IntrinsicStructuralSibling:
 			selectColumnIfNotAlready(columnPathSpanParentID)
 			continue
+
+		case traceql.IntrinsicNestedSetLeft:
+			nestedSetLeftExplicit = true
```

Comment:
this works b/c the columns are stored in the `columnPredicates` and `columnSelectAs` maps. 

if the explicit intrinsics hit first then they are added to the map and then the logic in `selectColumnIfNotAlready` prevents them from being overwritten with an `OpNone` condition.

if the structural intrinsics hit first then they are overwritten when the explicit intrinsics come along.

Reply:
i did note and fix an issue in this area. if the structural intrinsic were to be added after the nested set intrinsic then it would have seen a non-empty predicate and not added the nil predicate. added to test for nil predicate

https://github.com/grafana/tempo/pull/3497/files#diff-7f3f53f3d2f4e723f54f23778f62dfa089a61ad39f1c90ca4bf63a335aacf3a5R1144